Runtime 6.0.1 release notes (#1263)
astronomer/docs: software_versioned_docs/version-0.29/runtime-release-notes.md

For instructions on how to upgrade, read [Upgrade Airflow on Astronomer Software](manage-airflow-versions.md). For general product release notes, go to [Software release notes](release-notes.md). If you have any questions or a bug to report, reach out to [Astronomer support](https://support.astronomer.io).

## Astro Runtime 6.0.1

- Release date: September 26, 2022
- Airflow version: 2.4.0

### Bug fixes 

- Fixed an issue where Astro users could not access task logs on Deployments using Runtime 6.0.0

> - Backported a fix to an issue where logs were not loading from Celery workers ([#26493](https://github.com/apache/airflow/pull/26493))

Reviewer: ```suggestion
- Backported a fix to correct an issue where logs were not loading from Celery workers ([#26493](https://github.com/apache/airflow/pull/26493))
```

- Fixed [CVE-2022-40674](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-40674)

## Astro Runtime 5.0.9

- Release date: September 20, 2022
- Airflow version: 2.3.4

### Backported fixes from Apache Airflow 2.4

- Fixed an issue where logs were not loading from Celery workers ([#26337](https://github.com/apache/airflow/pull/26337) and [#26493](https://github.com/apache/airflow/pull/26493))
- Fixed CVE-2022-40754 ([#26409](https://github.com/apache/airflow/pull/26409))
- Fixed the Airflow UI not auto-refreshing when scheduled tasks are running. This bug was introduced in Airflow 2.3.4 ([#25950](https://github.com/apache/airflow/pull/25950))

> - Fixed an issue where the scheduler could crash when queueing dynamically mapped tasks ([#25788](https://github.com/apache/airflow/pull/25788))

Reviewer: ```suggestion
- Fixed an issue where the scheduler could become unresponsive when queueing dynamically mapped tasks ([#25788](https://github.com/apache/airflow/pull/25788))
```
![image](https://user-images.githubusercontent.com/104205257/192516050-f2e8b144-8963-4d17-827f-037193b06186.png)

### Additional improvements

> - Set `AIRFLOW__CELERY__STALLED_TASK_TIMEOUT=600` by default. This means that tasks that are in `queued` state for more than 600 seconds (10 minutes) will fail. This environment variable can be overridden on Astro but will help prevent tasks from getting stuck in a queued state.

Reviewer: ```suggestion
- Set `AIRFLOW__CELERY__STALLED_TASK_TIMEOUT=600` by default. Tasks that are in `queued` state for more than 600 seconds (10 minutes) fail. You can override this environment variable on Astro, but it helps prevent tasks from remaining in a queued state indefinitely.
```

> - Upgraded `astronomer-providers` to 1.8.1, which includes various bug fixes. For a complete list of changes, see the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/main/CHANGELOG.rst#181-2022-09-01).

Reviewer: ```suggestion
- Upgraded `astronomer-providers` to 1.8.1, which includes various bug fixes. See the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/main/CHANGELOG.rst#181-2022-09-01).
```

- Automatic DAG registration. You no longer need to specify `as dag` when defining a DAG object.
- Support for [zipping](https://airflow.apache.org/docs/apache-airflow/2.4.0/concepts/dynamic-task-mapping.html#combining-upstream-data-aka-zipping) dynamically mapped tasks.

## Astro Runtime 5.0.8

- Release date: August 23, 2022
- Airflow version: 2.3.4

### Support for Airflow 2.3.4

Astro Runtime 5.0.8 includes Airflow 2.3.4, which primarily includes bug fixes. For a complete list of commits, see the [Apache Airflow 2.3.4 release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-3-4-2022-08-23).

### Additional improvements

- Upgraded `astronomer-providers` to version `1.8.0`, which includes minor bug fixes and performance enhancements. For more information, see the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/1.8.0/CHANGELOG.rst).

> - Upgraded `openlineage-airflow` to version `0.13.0`, which includes support for Azure Cosmos DB. For a list of all changes, see the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).

Reviewer: ```suggestion
- Upgraded `openlineage-airflow` to version `0.13.0`, which includes support for Azure Cosmos DB. For more information, see the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).
```

## Astro Runtime 5.0.7

- Release date: August 16, 2022
- Airflow version: 2.3.3

### Backported fixes from Apache Airflow

Astro Runtime 5.0.7 includes the following bug fixes:

- Fixed an issue where [plugins specified as a python package](https://airflow.apache.org/docs/apache-airflow/stable/plugins.html#plugins-as-python-packages) in an `entry_points` argument were incorrectly loaded twice by Airflow and resulted in an error in the Airflow UI. This bug was introduced in Airflow 2.3.3 ([#25296](https://github.com/apache/airflow/pull/25296))
- Fixed an issue where zombie tasks were not properly cleaned up from DAGs with parse errors [#25550](https://github.com/apache/airflow/pull/25550))
- Fixed an issue where clearing a deferred task instance would not clear its `next_method` field ([#23929](https://github.com/apache/airflow/pull/23929))

These changes were backported from Apache Airflow 2.3.4, which is not yet generally available. The bug fixes were also backported to Astro Runtime 5.0.5.

### Additional improvements

> - Upgraded `openlineage-airflow` to version `0.12.0`, which includes support for Spark 3.3.0 and Apache Flink. For a list of all changes, see the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).

Reviewer: ```suggestion
- Upgraded `openlineage-airflow` to version `0.12.0`, which includes support for Spark 3.3.0 and Apache Flink. For more information, see the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).
```
